### ISSUE TITLE
Induction fails with strange warning and error messages when there are unconventionally named schemes 

### ISSUE THREAD

**Issue by JasonGross** [coq/coq#10766]
#### Description of the problem

```coq
Unset Elimination Schemes.

Inductive Nat := zero | succ (n : Nat).
Inductive Nat' := zero' | succ' (n : Nat').

Goal forall (P : Nat -> Type) n, P n.
  intros P n.
  Fail induction n. (* As expected: Error:
Cannot find the elimination combinator Nat_rect, the elimination of the inductive definition Nat on
sort Type is probably not allowed. *)
  Scheme Nat_ind := Induction for Nat Sort Type.
  Fail induction n. (* Still fails :-( *)
  Scheme Nat_rect := Induction for Nat Sort Prop.
  Fail induction n. (* Very confusing: The command has indeed failed with message:
Cannot instantiate metavariable P of type "Nat -> Prop" with abstraction
"fun n : Nat => P n" of incompatible type "Nat -> Type". *)
Abort.

Goal forall (P : Nat' -> Type) n, P n.
  intros P n.
  Scheme Nat'_rect := Minimality for Nat' Sort Type.
  induction n. (* Warning: Cannot remove n. [cannot-remove-as-expected,tactics] *)
  (* This should be an error here, and not leave the goal in a strange half-inducted state *)
```

I think that what `induction` should do is first look for schemes of any name generated by the `Scheme` command which are compatible (so, if your goal is in `Prop` and you are not dependent over the variable, it should first look for `Minimality` in `Sort Prop`, then `Induction` in `Sort Prop`, then `Minimality` in `Sort Set`, then `induction` in `Sort Set`, etc, but if you have a goal in `Type` and you are dependent over the variable, then you should only look for `Induction` in `Sort Type`).  (I could be sold on `induction` not doing a lookup like this, and instead only having one scheme that it checks: `Induction` for the sort of the goal.  But it should still look at things registered by `Scheme`, I believe.)  If no such induction scheme is found, then `induction` should fall back on the relevant default name (because users might want to define their own), *and* the type should be checked before we start doing induction.  Certainly, I should never see "Warning: Cannot remove n. [cannot-remove-as-expected,tactics]" from `induction`.

#### Coq Version

8.9.0

**Comment by jdchristensen** [comment]
I agree that something like Jason describes would be very helpful.  At the very least:  (1) allow induction schemes of sort `Type` to be used with a goal of sort `Set` (since `Set` is `Type@{Set}`), and (2) if an induction scheme is created with the `Scheme` command, remember the name and use that name in the appropriate context.

**Comment by SkySkimmer** [comment]
Should work now (we still get cannot remove warning when trying to do dependent induction with a non dependent scheme but IMO that's a separate issue)
